go-controller/pkg/crd/egressfirewall/v1/types.go

```diff
@@ -0,0 +1,78 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EgressNetworkFirewallRuleType indicates whether an EgressNetworkFirewallRule allows or denies traffic
+type EgressFirewallRuleType string
+
+const (
+	EgressFirewallRuleAllow EgressFirewallRuleType = "Allow"
+	EgressFirewallRuleDeny  EgressFirewallRuleType = "Deny"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=egressfirewall
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// EgressFirewall describes the current egress firewall for a Namespace.
+// Traffic from a pod to an IP address outside the cluster will be checked against
+// each EgressFirewallRule in the pod's namespace's EgressFirewall, in
+// order. If no rule matches (or no EgressFirewall is present) then the traffic
+// will be allowed by default.
+type EgressFirewall struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the desired behavior of EgressFirewall.
+	Spec EgressFirewallSpec `json:"spec"`
+	// Observed status of EgressFirewall
+	// +optional
+	Status EgressFirewallStatus `json:"status"`
+}
+
+// EgressFirewallStatus describes the lifecycle status of EgressFirewall.
+type EgressFirewallStatus struct {
+}
+
+// EgressFirewallSpec is a desired state description of EgressFirewall.
+type EgressFirewallSpec struct {
+	// a collection of egress firewall rule objects
+	Rules []EgressFirewallRule `json:"rules"`
+}
+
+// EgressFirewallRule is a single egressfirewall rule object
+type EgressFirewallRule struct {
+	// type marks this as an "Allow" or "Deny" rule
+	Type EgressFirewallRuleType `json:"type"`
+	// protocol marks what protocol for the rule to affect
+	EgressFirewallProtocol `json:"protocoldetails,omitempty"`
+	// to is one or more targes that traffic is allowed/denied to
+	To []Destination `json:"to"`
```

Comment:
To []EgressFirewallDestination `json:"to"`

Reply:
changed the struct name